sail/src/main/java/org/apache/rya/rdftriplestore/inference/HasSelfVisitor.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.rya.rdftriplestore.inference;
+
+import org.apache.rya.api.RdfCloudTripleStoreConfiguration;
+import org.openrdf.model.Resource;
+import org.openrdf.model.URI;
+import org.openrdf.model.vocabulary.RDF;
+import org.openrdf.query.algebra.Extension;
+import org.openrdf.query.algebra.ExtensionElem;
+import org.openrdf.query.algebra.StatementPattern;
+import org.openrdf.query.algebra.Var;
+
+/**
+ *
+ */
+public class HasSelfVisitor extends AbstractInferVisitor {
+    private static final Var TYPE_VAR = new Var("p", RDF.TYPE);
+    public HasSelfVisitor(final RdfCloudTripleStoreConfiguration conf, final InferenceEngine inferenceEngine) {
+        super(conf, inferenceEngine);
+        include = conf.isInferInverseOf();
```

Comment:
I think it's probably best to make them configurable.  We could just make the default returned by the config getter method be true.  But it provides a way of prioritizing some rules over others.